batch/create/create_persistent_disk_job.js

```diff
@@ -0,0 +1,156 @@
+        '-c',
+        'echo Hello world! This is task ${BATCH_TASK_INDEX}.' +
+          '>> /mnt/disks/NEW_PERSISTENT_DISK_NAME/output_task_${BATCH_TASK_INDEX}.txt',
+      ],
+    }),
+  });
+
+  // Define volumes and their parameters to be mounted to a VM.
+  const newVolume = new batch.Volume({
+    deviceName: newPersistentDiskName,
+    mountPath: `/mnt/disks/${newPersistentDiskName}`,
+    mountOptions: ['rw', 'async'],
+  });
+
+  const existingVolume = new batch.Volume({
+    deviceName: existingPersistentDiskName,
+    mountPath: `/mnt/disks/${existingPersistentDiskName}`,
+  });
+
+  const task = new batch.TaskSpec({
+    runnables: [runnable],
+    volumes: [newVolume, existingVolume],
+    maxRetryCount: 2,
+    maxRunDuration: {seconds: 3600},
+  });
+
+  // Tasks are grouped inside a job using TaskGroups.
+  const group = new batch.TaskGroup({
+    taskCount: 3,
+    taskSpec: task,
+  });
+
+  const newDisk = new batch.AllocationPolicy.Disk({
+    type: newDiskType,
+    sizeGb: diskSize,
+  });
+
+  // Policies are used to define on what kind of virtual machines the tasks will run on.
+  // Read more about local disks here: https://cloud.google.com/compute/docs/disks/persistent-disks
+  const instancePolicy = new batch.AllocationPolicy.InstancePolicy({
+    disks: [
+      // Create configuration for new disk
+      new batch.AllocationPolicy.AttachedDisk({
+        deviceName: newPersistentDiskName,
+        newDisk,
+      }),
+      // Create link to existing disk
+      new batch.AllocationPolicy.AttachedDisk({
+        existingDisk: `projects/${projectId}/${location}/disks/${existingPersistentDiskName}`,
+        deviceName: existingPersistentDiskName,
+      }),
+    ],
+  });
+
+  const locationPolicy = new batch.AllocationPolicy.LocationPolicy({
+    allowedLocations: [location],
+  });
+
+  const allocationPolicy = new batch.AllocationPolicy.InstancePolicyOrTemplate({
+    instances: [{policy: instancePolicy}],
+    location: locationPolicy,
+  });
+
+  const job = new batch.Job({
+    name: jobName,
+    taskGroups: [group],
+    labels: {env: 'testing', type: 'script'},
+    allocationPolicy,
+    // We use Cloud Logging as it's an option available out of the box
+    logsPolicy: new batch.LogsPolicy({
+      destination: batch.LogsPolicy.Destination.CLOUD_LOGGING,
+    }),
+  });
+  // The job's parent is the project and region in which the job will run
+  const parent = `projects/${projectId}/locations/${region}`;
+
+  async function callCreateBatchPersistentDiskJob() {
+    // Construct request
+    const request = {
+      parent,
+      jobId: jobName,
+      job,
+    };
+
+    // Run request
+    const [response] = await batchClient.createJob(request);
+    console.log(JSON.stringify(response));
+  }
+
+  callCreateBatchPersistentDiskJob();
+  // [END batch_create_persistent_disk_job]
+}
+
+process.on('unhandledRejection', err => {
+  console.error(err.message);
+  process.exitCode = 1;
+});
+
+main();
```

Comment:
Promise returned from main is ignored. You can address both this and the previous comment I made by make the following change:

```javascript
main().catch(err => {
  console.error(err);
  process.exitCode = 1;
});
```